app/auth/auth_service.ts

```diff
@@ -67,14 +68,19 @@ export class AuthService {
         this.refreshUser();
       })
       .catch((error: any) => {
-        this.onUserRpcError(error);
+        // TODO(siggisim): Remove "No user token" string matching after the next release.
+        if (BuildBuddyError.parse(error).code == "Unauthenticated" || error.includes("No user token")) {
```

Comment:
Since I'm a typescript baby -- what's the reason for using BuildBuddyError instead of a status object here? (i went down this road because I was going to ask why you compared the code with a string instead of an enum / const)

Reply:
The server currently (and I'd love to fix this at some point) returns http code `500` for all errors with just a text response that looks like this `rpc error: code = PermissionDenied desc = User not found`.

There's no status proto for us to parse, or http code for us to check. Changing this will require figuring out why the server isn't returning the right http codes (or returning status protos everywhere) and updating all of the places we parse errors to support both the current implementation and the new one.

Our error parser parses the current error format into an error code and an error description. 

To answer your actual question though, the error code looks like a string, but it's actually effectively an enum (the Typescript compiler will complain if it's an invalid value) thanks to this [union type](https://github.com/buildbuddy-io/buildbuddy/blob/master/app/util/errors.ts#L1) (@bduffany likes these, but I prefer enums).

Reply:
I recently learned that enums in TypeScript can be used the same way as those string unions, and I agree they're easier to understand -- going to switch to enums from now on :)